lib/LaTeXML/Core/Parameters.pm

```diff
@@ -60,10 +60,8 @@ sub readArguments {
   my ($self, $gullet, $fordefn) = @_;
   my @args = ();
   $gullet->setup_scan();
-  foreach my $parameter (@$self) {
-    my $value = $parameter->read($gullet, $fordefn);
-    push(@args, $value) unless $$parameter{novalue}; }
-  return @args; }
+  my ($p, $v);
+  return map { $p = $_; $v = $p->read($gullet, $fordefn); ($$p{novalue} ? () : $v); } @$self; }
```

Comment:
is this map _also_ faster than foreach? Should I entirely revise my rule of thumb and start treating map as faster? Shocking if so.

Reply:
Seems that it was. But then I was benchmarking on a windy day, so who knows?